src/search-modal/index.js

```diff
@@ -0,0 +1,3 @@
+// @ts-check
```

Comment:
Just make this file `.ts` please. Or do we really need this file?

Reply:
I think this is good for helping us avoid inter-feature import. We should not import from `./seach-modal/Something`. We should always import from explicitly exported members from `./search-modal`